### DISCUSSION TITLE
VRCX Integration

### DISCUSSION THREAD

**Discussion by Bluscream** [AdGoBye/AdGoBye#26] [General]
Why make yet another parser for my game logs, if you could just give that task to VRCX using it's IPC

**Comment by regalialong** [comment]
VRCX IPC is undocumented ~~so I wasn't aware it could do that~~.

Either way, I don't know if doing IPC with VRCX is possible on Linux, VRCX on Wine already requires a custom build and a lot of sanity. You're free to create your own fork and replace the log parser in Live though.

**Reply by DubyaDude** [reply]
It is not possible with VRCX over IPC in its current codebase (it would require some changes on VRCXs end).

Not only does that technical hurdle exist, but VRCX in general is much more game TOS-friendly than this project, so there would be a massive implication there as well.

**Reply by Bluscream** [reply]
Interesting. I didnt know AGB was possible on wine either since the cache paths are window. Also i have to admit that its a shame that VRVXs IPC stuff is undocumented either, i only found out because of Just H's yoinkerdetector. I hoped it would give events for OnItemCached or smth along those lines. Thanks for clearing this up. I wonder how vrcx is so hard to run on linux even though its just "yet another embedded web browser".

However, despite all this, i would still see some usefulness in optionally embedding vrcxs ipc stuff for ingame notifications (example: "Current world has been patched, rejoin to apply...", etc)
Also about the whole ToS-compability stuff; i would argue thats the whole point behind vrcx ipx, so that projects like yoinkerdetector can exist, which is probably also not in line with the games tos

**Reply by Myrkie** [reply]
VRCX IPC integration was originally made for use with the VRCX Companion mod, that mod not longer exists so its just kept as a legacy, if apps wish to use it they can but its functionality is limited to its original purpose (sending messages to VRCX) if you want to use the IPC client you can refer to yoinker detector as an unofficial example https://github.com/aclyptos/StandaloneNotifer

this being said VRCX has no control over what the IPC client is used for only its functionality (which is no longer maintained), malicious or not but making changes to VRCX to directly assist another application with malicious or TOS unfriendly activities will not be assisted, yoinker detector just uses what was available without any input from VRCX maintainers

**Reply by regalialong** [reply]
> Interesting. I didnt know AGB was possible on wine either since the cache paths are window. [sic; Windows?]

We compile AGB to Linux so we don't require Wine for this, but VRCX requires Wine since it doesn't have a Linux build.

The crux of what I was trying to say is that I don't think we can do IPC from Linux to VRCX running in Wine (or if we can, it'd be probably cursed).

> I wonder how vrcx is so hard to run on linux even though its just "yet another embedded web browser".

It requires a setup environment and a custom build, so it's a bit of a hassle. Why? Technical debt would be your answer I guess, off topic here though.

> ingame notifications

I've had that planned but my focus has been on internals, I think this makes sense to implement when there's some form of dependency injection available (there isn't right now).

**Reply by Bluscream** [reply]
gotcha. Btw, i found the source of the yoinker detector. https://github.com/aclyptos/StandaloneNotifer

And yeah it has it's own logparser.

Now i have 3 game logparsers running at the same time while playing, great way to tank my performance i guess :joy:

**Reply by Myrkie** [reply]
> VRCX IPC integration was originally made for use with the VRCX Companion mod, that mod not longer exists so its just kept as a legacy, if apps wish to use it they can but its functionality is limited to its original purpose (sending messages to VRCX) if you want to use the IPC client you can refer to yoinker detector as an unofficial example https://github.com/aclyptos/StandaloneNotifer
> 
> this being said VRCX has no control over what the IPC client is used for only its functionality (which is no longer maintained), malicious or not but making changes to VRCX to directly assist another application with malicious or TOS unfriendly activities will not be assisted, yoinker detector just uses what was available without any input from VRCX maintainers


if you scroll up I said this

**Comment by Myrkie** [comment]
adding to this: VRCX IPC doesn't have this functionality it cannot pass log parsing events over IPC thus would also require changes made to VRCX.

**Comment by DubyaDude** [comment]
To provide a concise answer to finish the discussion.

VRCX support will not be natively added for 'less reads of game logs' as that's not how VRCX IPC works. We will not be pushing VRCX to update a deprecated feature for this due to the game's TOS reasons.

Applications doing Read Only on a log file that doesn't do a write lock whatsoever does not noticeably impact the performance of the writer or reader(s), modern OSs are meant to handle this sort of situation.

You'll need to create your own forks of these if it matters to you this much.
